compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -410,23 +412,34 @@ object Completion {
     val Import: Mode = new Mode(4) | Term | Type
   }
 
-  /** A scope that tracks renames of the entered symbols.
-   *  Useful for providing completions for renamed symbols
-   *  in the REPL and the IDE.
-   */
-  private class RenameAwareScope extends Scopes.MutableScope {
-    private val nameToSymbols: mutable.Map[TermName, List[Symbol]] = mutable.Map.empty
-
-    /** Enter the symbol `sym` in this scope, recording a potential renaming. */
-    def enter[T <: Symbol](sym: T, name: Name)(using Context): T = {
-      val termName = name.stripModuleClassSuffix.toTermName
-      nameToSymbols += termName -> (sym :: nameToSymbols.getOrElse(termName, Nil))
-      newScopeEntry(name, sym)
-      sym
-    }
+  private object CompletionScope  {
+    val empty = CompletionScope()
+  }
+
+  private case class CompletionScope(nameToDenots: Map[Name, List[SingleDenotation]] = Map.empty) {
```

Comment:
`CompletionScope` was supposed to be a lightweight wrapper over `Map` taking care of filtering out not matching completions and resolving conflicts between different denotations with the same name.
`CompletionBuffer` holds much more logic

Reply:
What about keeps a mutable map inside `CompletionBuffer`, which is closer to what the name `buffer` suggests.

Reply:
To clarify, I'm not proposing the change. I'm just trying to understand the design considerations here.

Reply:
The name comes from the times when it indeed was mutable but some corner cases seemed too hard to fix with mutability. Any idea for a better name?

Reply:
I'm not familiar with this part of code, so I'm asking questions to learn more about it.

I think the name is OK, I'm more interested in learning the relationships and responsibilities of the classes.

Creating a separate class with only one field looks dubious. However, in this case it may be justified if there is some special semantic operations defined around for this class.